src/components/Dropdown/SortDropdown/useSortDropdown.ts

```diff
@@ -0,0 +1,118 @@
+import {
+  ChangeEventHandler,
+  ComponentProps,
+  MouseEventHandler,
+  useCallback,
+  useMemo,
+  useState,
+} from 'react'
+
+import { FaSortAmountDownIcon, FaSortAmountUpIcon } from '../../Icon'
+
+import { SortDropdown } from './SortDropdown'
+import { sortDropdownStyle } from './style'
+
+type Props = Omit<ComponentProps<typeof SortDropdown>, 'onCancel'>
+
+const SORT_FIELD_LABEL = '並び替え項目'
+const SORT_ORDER_LABEL = '並び順'
+const ASC_LABEL = '昇順'
+const DESC_LABEL = '降順'
+const APPLY_BUTTON_TEXT = '適用'
+const CANCEL_BUTTON_TEXT = 'キャンセル'
+
+export const useSortDropdown = ({ sortFields, defaultOrder, onApply, decorators }: Props) => {
+  const sortFieldLabel = useMemo(
+    () => decorators?.sortFieldLabel?.(SORT_FIELD_LABEL) || SORT_FIELD_LABEL,
+    [decorators],
+  )
+  const sortOrderLabel = useMemo(
+    () => decorators?.sortOrderLabel?.(SORT_ORDER_LABEL) || SORT_ORDER_LABEL,
+    [decorators],
+  )
+  const ascLabel = useMemo(() => decorators?.ascLabel?.(ASC_LABEL) || ASC_LABEL, [decorators])
+  const descLabel = useMemo(() => decorators?.descLabel?.(DESC_LABEL) || DESC_LABEL, [decorators])
+  const applyButtonLabel = useMemo(
+    () => decorators?.applyButtonLabel?.(APPLY_BUTTON_TEXT) || APPLY_BUTTON_TEXT,
+    [decorators],
+  )
+  const canselButtonLabel = useMemo(
+    () => decorators?.canselButtonLabel?.(CANCEL_BUTTON_TEXT) || CANCEL_BUTTON_TEXT,
+    [decorators],
+  )
+
+  // 外向きの値
+  const [selectedField, setSelectedField] = useState<string>()
+  const [checkedOrder, setCheckedOrder] = useState<Props['defaultOrder']>(defaultOrder)
```

Comment:
nits) ここのラベル用の変数と内部の状態管理用の変数が、少し理解するのに負荷があるように感じました。
用途としてはラベル用でしかないため、 `Field` を `Label` にして用途が想像できるようにするのはいかがでしょう？

```suggestion
  // 外向きの値
  const [selectedLabel, setSelectedLabel] = useState<string>()
  const [checkedOrder, setCheckedOrder] = useState<Props['defaultOrder']>(defaultOrder)
```

Reply:
typo と合わせて修正しました!!!!!!!! https://github.com/kufu/smarthr-ui/pull/4301/commits/182bd6479b699b9697d6da0ca2af3e24cb963712

Reply:
ありがとうございます！とてもわかりやすくなりました :+1: